core/modules/replica/ChunkLocker.h

```diff
@@ -0,0 +1,222 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_CHUNK_LOCKER_H
+#define LSST_QSERV_REPLICA_CHUNK_LOCKER_H
+
+/// ChunkLocker.h declares:
+///
+/// struct Chunk
+/// class  ChunkLocker
+///
+/// (see individual class documentation for more information)
+
+// System headers
+#include <list>
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+
+// Qserv headers
+#include "util/Mutex.h"
+
+// Forward declarations
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Structure Chunk is an abstraction groupping together database families and
+ * chunk numbers. This is needed to support chunk replication operations which
+ * require chunk collocation.
+ */
+struct Chunk {
+    std::string databaseFamily;
+    unsigned int number;
```

Comment:
`chunkNumber` would be more descriptive, which may be helpful to anyone unfamiliar with the code.

Reply:
I thought this association is already implied by the class name :-) Besides, consider this example:

Chunk chunk{"family", 123};

if (chunk.chunkNumber == 123) ...

versus:

if (chunk.number == 123) ...

The second looks better (from my personal point of view).

At the mean time, there may be a problem with the class name (Chunk) as I have a few different contexts where I'm using this word ("chunk"). Personally, I'm not quite happy with some naming choices I made in the code. Let's talk on this tomorrow (Friday). It would be easier to explain this in person.

Reply:
you're assuming I know that chunks are numbered. IMO int number is pretty mean 😆.
you could make it a class & have private data members, and then have your accessor function `number()`, and let the data member be named `chunkNumber`. That way you still get `chunk.number() == 123`.

Reply:
(I'm on PTO tomorrow, camping :tent:)